src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -98,13 +98,24 @@ public function __construct(array $options = [])
         } elseif (isset($_SERVER['argv']) && class_exists(ArgvInput::class)) {
             $this->options = $options;
             $this->getInput();
+            $inputEnv = $_SERVER[$envKey] ?? null;
+            $inputDebug = $_SERVER[$debugKey] ?? null;
         }
 
         if (!($options['disable_dotenv'] ?? false) && isset($options['project_dir']) && !class_exists(MissingDotenv::class, false)) {
             (new Dotenv($envKey, $debugKey))
                 ->setProdEnvs((array) ($options['prod_envs'] ?? ['prod']))
                 ->usePutenv($options['use_putenv'] ?? false)
-                ->bootEnv($options['project_dir'].'/'.($options['dotenv_path'] ?? '.env'), 'dev', (array) ($options['test_envs'] ?? ['test']), $options['dotenv_overload'] ?? false);
+                ->bootEnv($options['project_dir'].'/'.($options['dotenv_path'] ?? '.env'), 'dev', (array) ($options['test_envs'] ?? ['test']), $dotenvOverload = $options['dotenv_overload'] ?? false);
+            if ($dotenvOverload) {
+                if (isset($inputEnv) && $inputEnv !== $_SERVER[$envKey]) {
+                    putenv($envKey.'='.$_SERVER[$envKey] = $_ENV[$envKey] = $inputEnv);
```

Comment:
we should throw a LogicException here instead beause it means there is a conflict between what `.env` declares as the env, and what `--env`. At this stage, the conflict means that Dotenv loaded the wrong `.env.$env` file if any.